multi-arch-builders/coreos-s390x-rhcos-builder.bu

```diff
@@ -91,3 +102,54 @@ storage:
 
             echo "Importing tarball into volume"
             sudo -u builder -H /bin/bash -c "cd /var/home/builder; podman volume import secex-data /var/home/builder/${TARBALL}"
+
+    - path: /usr/local/bin/create-cex-config.sh
+      mode: 0755
+      user:
+        name: root
+      group:
+        name: root
+      contents:
+        inline: |
+            #!/bin/bash
+            set -e
+
+            sudo modprobe vfio_ap
+            UUID='68cd2d83-3eef-4e45-b22c-534f90b16cb9'
+            if  ! ls /sys/devices/vfio_ap/matrix | grep -q "${UUID}"; then
+                cards=$(ls /sys/bus/ap/devices | grep -E 'card[0-9]{2}')
+                if [ -z "${cards}" ]; then
+                    echo "cex device not found."
+                    exit 1
+                fi
+
+              # if more than one card picks the first available cca controller
+              for card in $cards; do
+                  if $(grep -q cca /sys/bus/ap/devices/${card}/uevent); then
+                      card=${card}
+                      card_domain=$(ls /sys/bus/ap/devices/$card/ | grep -E '[0-9a-f]{2}.[0-9a-f]{4}')
+                      break
+                  fi
+              done
```

Comment:
Should check here if we have a card and card_domain and error out if not.

Reply:
Hi @jlebon , 
Once the domain is assigned to the LPAR level , then the domain will remain available under the /sys/devices/ap/\<cardxx\>/ folder. During the builder LPAR cex configuration, they had assigned only card (not sure that was the process of assigning the card)  and not domain, and OS was not able to detect the card. Later the card and domain is visible only the moment they had assigned the domain to the lpar configuration. 

Kindly suggest..


 Here is the output from the builder lpar.  I've done similar verification at the end of the script but after configuring the mediated device.

```
core@coreos-s390x-builder:~$ ls /sys/devices/ap/
card00  uevent

core@coreos-s390x-builder:~$ ls /sys/devices/ap/card00/
00.0014       chkstop  depth   hwtype  max_msg_size  online          raw_hwtype     requestq_count  subsystem  uevent
ap_functions  config   driver  load    modalias      pendingq_count  request_count  serialnr        type

core@coreos-s390x-builder:~$ ls -l /sys/bus/ap/devices
total 0
lrwxrwxrwx. 1 root root 0 Jun 24 07:06 00.0014 -> ../../../devices/ap/card00/00.0014
lrwxrwxrwx. 1 root root 0 Jun 24 07:06 card00 -> ../../../devices/ap/card00
```

Reply:
What I mean here is that you have a for-loop just above that looks for the card and assigns to `card` and `card_domain` and breaks out of the loop. We should verify that those variables were assigned before using them further down. E.g.

```bash
if [ -z "$card" ] || [ -z "$card_domain" ]; then
    echo "couldn't find card with CCA controller"
    exit 1
fi
```

Reply:
Ok, Will do that.